cli/file_watcher.rs

```diff
@@ -18,22 +18,22 @@ use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
 use std::time::Duration;
 use tokio::select;
-use tokio::time::{interval, Interval};
+use tokio::time::{delay_for, Delay};
 
 const DEBOUNCE_INTERVAL_MS: Duration = Duration::from_millis(200);
 
 // TODO(bartlomieju): rename
-type WatchFuture = Pin<Box<dyn Future<Output = Result<(), AnyError>>>>;
+type WatchFuture<T> = Pin<Box<dyn Future<Output = Result<T, AnyError>>>>;
```

Comment:
Let's address this TODO, how about `FileWatcherFuture`?

Reply:
sounds good 👍